pymatviz/utils.py

```diff
@@ -299,23 +299,24 @@ def annotate_metrics(
 
 def crystal_sys_from_spg_num(spg: int) -> CrystalSystem:
     """Get the crystal system for an international space group number."""
-    # not using isinstance(n, int) to allow 0-decimal floats
```

Comment:
Should the case of `spg` not being `int` even be considered here? As the type hint already explicitly requires `type(spg)` to be `int`.

Reply:
Type hints can be ignored and it's nice to give specific error messages even then. Often people feed in `pandas` dataframes columns and aren't aware what datatype they have.

Reply:
I usually do one of two things:

1. Assume the ideal case (right data type is passed), usually for priviate func/method not accessed by the user directly. Which might make the code looks clean but risk the robustness.

2. Assume the worst scenario where any irrational input could be passed in, and implement a strict checking (type, value and so on).

In the original code:
https://github.com/janosh/pymatviz/blob/882a6caaa06acbcadbcfc3d8e2d7fe6aa8ce630e/pymatviz/utils.py#L302-L304

This seems to be standing somewhere in the middle. Maybe I should go ahead and switch to strick checking?

Reply:
> Maybe I should go ahead and switch to strict checking?

Certainly, stricter checks and better error messages are always welcome.

Reply:
Thanks for the input as always @janosh